AlphaWallet/Browser/Coordinators/ScanQRCodeCoordinator.swift

```diff
@@ -8,12 +8,59 @@ protocol ScanQRCodeCoordinatorDelegate: class {
     func didScan(result: String, in coordinator: ScanQRCodeCoordinator)
 }
 
-final class ScanQRCodeCoordinator: NSObject, Coordinator {
-    var coordinators: [Coordinator] = []
-    weak var delegate: ScanQRCodeCoordinatorDelegate?
+protocol ScanQRCodeCoordinatorResolutionDelegate: class {
+    func coordinator(_ coordinator: ScanQRCodeCoordinator, didResolveAddress address: AlphaWallet.Address, action: ScanQRCodeAction)
+    func coordinator(_ coordinator: ScanQRCodeCoordinator, didResolveWalletConnectURL url: WCURL)
+    func coordinator(_ coordinator: ScanQRCodeCoordinator, didResolveString value: String)
+    func coordinator(_ coordinator: ScanQRCodeCoordinator, didResolveURL url: URL)
+}
+
+enum ScanQRCodeAction: CaseIterable {
+    case sendToAddress
+    case addCustomToken
+    case watchWallet
+
+    var title: String {
+        switch self {
+        case .sendToAddress:
+            return R.string.localizable.qrCodeSendToAddressTitle()
+        case .addCustomToken:
+            return R.string.localizable.qrCodeAddCustomTokenTitle()
+        case .watchWallet:
+            return R.string.localizable.qrCodeWatchWalletTitle()
+        }
+    }
+}
+
+typealias WCURL = String
+
+private enum ScanQRCodeResolution {
+    case address(AlphaWallet.Address)
+    case walletConnect(WCURL)
+    case other(String)
+    case url(URL)
+
+    init(rawValue: String) {
+        if let address = AlphaWallet.Address(string: rawValue) {
```

Comment:
> Should have something like this:
> 
> ```
> let rawValue = rawValue.trimmed
> ```
> 
> so addresses like "0x89d24a6b4ccb1b6faa2625fe562bdd9a23260359\n" is handled. I don't know how I generated it, but I guess it might be common :)

done